### ISSUE TITLE
legacy_generator_methods_support: Legacy parameter convertion bug

### ISSUE THREAD

**Issue by SolessChong** [fchollet/keras#6210]
In legacy generator convertor, I'm sure that `val_samples` in ver. 1 is not equavalent to 'steps' in ver. 2. I suggest divede by batch_size when converting.

Howerver, it seems not able to get batch_size from generator interface. Can you guys suggest a way to fix this? I'm willing to contribute code.

    gen = ImageDataGenerator()
    train_batches = gen.flow_from_directory("data/train", model.input_shape[1:3], shuffle=False, batch_size=16)

    train_bottleneck = model.predict_generator(train_batches, train_batches.samples)

    # This predict_generator will iterate over dataset for N times where N is `train_batch.batch_size`.

while...

    train_bottleneck = model.predict_generator(train_batches, -(-train_batches.samples // train_batches.batch_size))  
    
    # This yeilds the right thing.
